### PR TITLE
Create App: Material Assistant: Add Mtoon, Shader_simple, missing PBR properties and bug fixes.

### PR DESCRIPTION
**Pull request by AleziaKurdis**
**This PR includes:**
 
- A bug fix regarding the Assistant panel that was closing for no reason on some very active domain.
 
- The X to close the assistant is now constantly present at the top of the panel. The page scrolls now under it. _(it was highly annoying to have to scroll up to close it.)_
 
- Add the support for all the existing models:  **vrm_mtoon** and **hifi_shader_simple** ( in addition to "hifi_pbr")

image

image
image

 
- Add the support for missing "**hifi_pbr**" properties: **lightMap**, and the 3 alternatives maps: **glossMap**, **bumpMap**, **specularMap**.
 
image

**Properties not supported:**
- **materialParams** _(Cause not used)_
- **lightmapParams** _(Cause not used)_
- **texCoordTransform0** _(Cause this is Mat4, not really human friendly.)_ 
- **texCoordTransform1** _(Cause this is Mat4, not really human friendly.)_ 
 
Note: The Material Assistant won't be available if "texCoordTransform0" and/or "texCoordTransform0" are present in the JSON. (to not affect the json)
 
**Mtoon properties not yet supported :** _(Despite that the doc say it is, it is actually not yet supported)_
-**outlineWidthMode**
-**outlineWidth**
-**outline**
**BUT**, it is coded and then put in comment in the code: See in **entityProperties.html** and in **materialAssistant.js**. So it will be easy to activate without have to re-dive deeply into that complex ui.)
 
 
This PR addresses the Issue #1076.

 
I have tested the generated JSON its reading.
But I have not tested the effect of those properties.
So test it carefully, cause it is a complex ui with a lot of exception and rules.
 
 
**Note:** If for any reason the json contains a map and its alternative one in the same time (Ex: normalMap and bumpMap), then the assistant will clean the inconsistency and keep only the last one in the json. (According the dev, this is what the code would use anyway if such json would be used.)

### CODE REVIEW

**Comment by JulianGro** [comment]
> Mtoon properties not yet supported : (Despite that the doc say it is, it is actually not yet supported)
> -outlineWidthMode
> -outlineWidth
> -outline

What document are you looking at? The outline support is supposed to be disabled.

**Comment by HifiExperiments** [comment]
I disabled them internally but they’re still in the Entities.Material docs.  I can add a note there about how they’re currently not implemented

File: scripts/system/create/entityProperties/html/js/materialAssistant.js [hunk 1]
```diff
@@ -389,27 +1015,157 @@ function initiateMaUi() {
...
+            maModelHifiPbr.checked = true;
+            break;
+        case "vrm_mtoon":
+            maModelVrmMtoon.checked = true;
+            break;
+        case "hifi_shader_simple":
+            maModelHifiShaderSimple.checked = true;
+            break;
+        default:
+            alert("ERROR: model = '" + maMaterialData.model + "'. Something has been broken in the code.");
```
Comment (ksuprynowicz) [inline scripts/system/create/entityProperties/html/js/materialAssistant.js:1167]: Maybe we should change "Something has been broken in the code" to "Unrecognized material type"?

**Review by ksuprynowicz — APPROVED** [review]
This is so awesome!
MToon material will get a lot more use now since it will be a lot easier to experiment with.
Everything in the code looks fine :)

File: scripts/system/create/entityProperties/html/js/materialAssistant.js (hunk 1 above)
Comment (AleziaKurdis) [reply scripts/system/create/entityProperties/html/js/materialAssistant.js]: I changed all the error messages of the assistant using the  following pattern:
**ERROR: Unrecognized material _{propertyName}_. (_{propertyName}_ = _{current value}_)**

**Comment by ksuprynowicz** [comment]
We did a lot of testing, especially during Maker Monday and everything seems to work perfectly. I think it's safe to merge now :)

**Pull request merged**